opts/opts_test.go

```diff
@@ -50,21 +50,28 @@ func TestNewOpts(t *testing.T) {
 
 	for _, tt := range tests {
 		got, err := NewOpts(tt.loaders...)
+		assert(t, err, tt.errType, got, tt.want)
+	}
+}
 
-		if tt.errType != nil {
-			if err == nil {
-				t.Fatal("expect an error")
-			}
+func assert(t *testing.T, gotErr, wantErrType error, got, want interface{}) {
```

Comment:
Nice, I really like this approach but I feel like this function deserves its own go file as it is used in multiple test files now.